Cython/Build/pep517_backend/_backend.py

```diff
@@ -0,0 +1,142 @@
+# Note - setuptools imports are all lazy so that we can specify it as a dependency
+from contextlib import contextmanager as _contextmanager
+from functools import lru_cache
+from pathlib import Path as _Path
+
+def _load_toml_file(filename: str) -> dict:
+    # I don't really want to be dealing with finding/vendoring replacement toml
+    # libraries for Python <3.11.  Therefore use setuptools if possible.
+    # Try a number of options for maximum chance of success.
+    try:
+        # pyprojecttoml is marked as private in setuptools
+        from setuptools.config.pyprojecttoml import load_file
+    except ImportError:
+        try:
+            # setuptools 69.1.0+
+            from setuptools.compat.py310 import tomllib
+        except ImportError:
+            import tomllib  # standard library from 3.11 onwards
+        def load_file(filename):
+            with open(filename, "rb") as f:
+                return tomllib.load(f)
+    return load_file(filename)
+
+
+def _prepare_ext_module(module_settings):
+    from setuptools import Extension as _Extension
+    name = module_settings['name']
+    sources = module_settings['sources']
+    
+    # TODO - everything else:
+    #  include_dirs, libraries, library_dirs, extra_*_args,
+    #  py_limited_api, etc., etc.
+    #  probably handle numpy/pythran/etc includes somehow?
+
+    return _Extension(name, sources)
+
+@lru_cache()  # incase we go through multiple calls to this function
```

Comment:
Why do you think it's called several times? PEP 517 hooks each are called in subprocesses IIUC.

Reply:
Yes - it looks like they are. In which case caching is pointless